components/observatorium.libsonnet

```diff
@@ -29,6 +29,22 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
         deduplicationReplicaLabels: obs.config.deduplicationReplicaLabels,
         deleteDelay: '48h',
       },
+      statefulSet+: {
+        spec+: {
+          template+: {
+            spec+: {
+              containers: [
+                if c.name == 'thanos-compact' then c {
+                  args+: [
```

Comment:
I think a slightly modified strategy might be better here, I think we should first filter any `--log.level=` arg out, and then add it again, so that if any such multiple merges are essentially idempotent.